### ISSUE TITLE
[Bug]: Unable to create server from snapshot in v1.38.0

### ISSUE THREAD

**Issue by apricote** [hetznercloud/terraform-provider-hcloud#656]
### What happened?

See https://github.com/hetznercloud/terraform-provider-hcloud/commit/2654d819f147dbe8367089d2dcea98d1e89d5564#r108628064

Creating a server with

```hcl
resource hcloud_server test {
  image = hcloud_snapshot.test.id
}
```

does not work

### What did you expect to happen?

Expected to still work when passing an Image/Snapshot ID

### Please provide a minimal working example

Below example expects a snapshot with architecture x86 and label `app=hcloud-example` to exist.

```hcl
data "hcloud_image" "snapshot" {
  with_architecture = "x86"
  with_selector = "app=hcloud-example"
}

resource "hcloud_server" "node" {
  name        = "tf-656"
  location    = "fsn1"
  server_type = "cx11"
  image       = data.hcloud_image.snapshot.id
}
```

**Comment by lars-hagen** [comment]
I actually made a slight mistake in my previous comment about the matter. 

> It just tries to
> `https://api.hetzner.cloud/v1/images?architecture=x86&include_deprecated=true&name=ID_HERE&page=1`
> Which wont list the snapshot and fails, the problem is that if you dont specify directly in the query above that you want snapshots `?type=snapshot` it will just not list the selected snapshot you want, and we cannot in the `hcloud_image` specify the image type we want is a snapshot.

`https://api.hetzner.cloud/v1/images?architecture=x86&include_deprecated=true&name=ID_HERE&page=1`

This will never work for snapshots as the name variable returned in the json is always null. So `name=ID_HERE`, will never work with `GET /images/` <https://docs.hetzner.cloud/#images>, even if you add `type=snapshot`. The only way to list snapshots with the query above is to remove `name` and add `type=snapshot`

 `https://api.hetzner.cloud/v1/images?architecture=x86&include_deprecated=true&type=snapshot&page=1` so this will list your snapshots, but you will never be able to specify which snapshot you want using `GET /images/`

![null](https://user-images.githubusercontent.com/1360677/231705688-6167d39f-1ec8-4d86-9f00-77e2b043517e.png)


The only way to get a snapshot is by directly querying it with `GET /image/ID_HERE` https://docs.hetzner.cloud/#images-get-an-image. so thats just what we need back.

To get snapshots working again I think we need to look at the code removed here, I gave up trying to setup this repo on my end but im quite sure the error is somewhere here. 

https://github.com/hetznercloud/terraform-provider-hcloud/commit/2654d819f147dbe8367089d2dcea98d1e89d5564#diff-7378c7d462221547688e0d724658c2450b4489a0b02abcd83ad57b30ffa530c4L273-L275

and 

https://github.com/hetznercloud/terraform-provider-hcloud/commit/2654d819f147dbe8367089d2dcea98d1e89d5564#diff-7378c7d462221547688e0d724658c2450b4489a0b02abcd83ad57b30ffa530c4L279-L291

**Comment by apricote** [comment]
Yes, that is correct :) The issue is that this was using `ImageClient.GetByNameAndArchitecture` which can only handle names instead of [`ImageClient.GetForArchitecture`](https://pkg.go.dev/github.com/hetznercloud/hcloud-go/hcloud) which correctly handles ids & names.

**Comment by mysticaltech** [comment]
I can confirm that this has been fixed in v1.38.1, thanks again @apricote for dealing with this promptly.
